### PR TITLE
chore(kubernetes_logs source): Document built-in metrics pipeline that we ship with the Helm charts

### LINKED ISSUE #5043: Document built-in metrics pipeline that we ship with the Helm charts
Each if the Helm charts provide an `internal_metrics` source and `prometheus` sink out of the box (#3799). Agents also expose `host_metrics` (#4163) via the same `prometheus` sink.

Charts come with options to enable Prometheus integration via `annotations` or Prometheus Operator integration via `PodMonitor`. 
We need to document this, and make it clear on how to enable these integrations, and what's the difference.

We also need to explain that the `node_exporter` is not required when the `host_metrics` is enabled.

### PR DESCRIPTION
**Pull request by eeyun**
Closes #5043 
Signed-off-by: Ian Henry <ianjhenry00@gmail.com>

### CODE REVIEW

**Comment by eeyun** [comment]
@binarylogic can you validate that I've put this in the right spot?

File: docs/reference/components/sources/kubernetes_logs.cue [hunk 1]
```diff
@@ -359,6 +359,20 @@ components: sources: kubernetes_logs: {
...
+		metrics: {
+			title: "Metrics"
+			body: """
+				Our Helm chart deployments provide QoL around setup and maintenance of
+				metrics pipelines in Kubernetes. Each of the Helm charts provide an internal_metrics
+				source and prometheus sink out of the box. Agent deployments also expose host_metrics
+				via the same prometheus sink.
+
+				Charts come with options to enable Prometheus integration via annotations or Prometheus Operator
+				integration via PodMonitor. Thus, node_exporter is nt required when the `host_metrics` are
```
Comment (jamtur01) [inline docs/reference/components/sources/kubernetes_logs.cue:371]: not

File: docs/reference/components/sources/kubernetes_logs.cue [hunk 2]
```diff
@@ -359,6 +359,20 @@ components: sources: kubernetes_logs: {
 				"""
 		}
 
+		metrics: {
+			title: "Metrics"
+			body: """
+				Our Helm chart deployments provide QoL around setup and maintenance of
+				metrics pipelines in Kubernetes. Each of the Helm charts provide an internal_metrics
```
Comment (jamtur01) [inline docs/reference/components/sources/kubernetes_logs.cue:366]: should `internal_metrics` and `prometheus` and `host_metrics` be code-formatted consistently?

Comment (eeyun) [reply docs/reference/components/sources/kubernetes_logs.cue:366]: yeah absolutely!

**Comment by binarylogic** [comment]
@eeyun can you put directly in here?

https://github.com/timberio/vector/blob/master/docs/reference/installation/platforms/kubernetes.cue#L12

since this is specific to the k8s platform, not the log sources.

**Comment by eeyun** [comment]
oh! yes absolutely @binarylogic !

**Review by jamtur01 — APPROVED** [review]
LGTM

File: docs/reference/installation/platforms/kubernetes.cue [hunk 3]
```diff
@@ -9,5 +9,21 @@ installation: platforms: kubernetes: {
...
+			title: "Metrics"
+			body: """
+				Our Helm chart deployments provide QoL around setup and maintenance of
+				metrics pipelines in Kubernetes. Each of the Helm charts provide an `internal_metrics`
+				source and `prometheus` sink out of the box. Agent deployments also expose `host_metrics`
+				via the same `prometheus` sink.
+
+				Charts come with options to enable Prometheus integration via annotations or Prometheus Operator
+				integration via PodMonitor. Thus, node_exporter is not required when the `host_metrics` are
+				enabled.
```
Comment (binarylogic) [inline docs/reference/installation/platforms/kubernetes.cue:25]:
```suggestion
				Our Helm chart deployments provide quality of life around setup and maintenance of
				metrics pipelines in Kubernetes. Each of the Helm charts provide an `internal_metrics`
				source and `prometheus` sink out of the box. Agent deployments also expose `host_metrics`
				via the same `prometheus` sink.

				Charts come with options to enable Prometheus integration via annotations or Prometheus Operator
				integration via PodMonitor. Thus, the Prometheus node_exporter agent is not required when the `host_metrics` source is
				enabled.
```

**Review by MOZGIII — APPROVED** [review]

**Pull request merged**